src/FSharp.Data.GraphQL.Server.AspNetCore/Giraffe/HttpHandlers.fs

```diff
@@ -1,155 +1,268 @@
-    /// do a custom handling on the response. For example, when you want
-    /// to add custom headers.
-    let handleGraphQLWithResponseInterception<'Root>
-        (cancellationToken : CancellationToken)
-        (logger : ILogger)
-        (interceptor : HttpHandler)
-        (next : HttpFunc)
-        (ctx : HttpContext)
-        =
-        task {
-            let cancellationToken =
-                CancellationTokenSource
-                    .CreateLinkedTokenSource(cancellationToken, ctx.RequestAborted)
-                    .Token
-            if cancellationToken.IsCancellationRequested then
-                return (fun _ -> None) ctx
-            else
-                let options = ctx.RequestServices.GetRequiredService<GraphQLOptions<'Root>> ()
-                let executor = options.SchemaExecutor
-                let rootFactory = options.RootFactory
-                let serializerOptions = options.SerializerOptions
-                let deserializeGraphQLRequest () = task {
-                    try
-                        let! deserialized = JsonSerializer.DeserializeAsync<GraphQLRequest> (ctx.Request.Body, serializerOptions)
-                        return Ok deserialized
-                    with :? GraphQLException as ex ->
-                        logger.LogError (``exception`` = ex, message = "Error while deserializing request.")
-                        return Result.Error [ $"%s{ex.Message}\n%s{ex.ToString ()}" ]
-                }
 
-                let applyPlanExecutionResult (result : GQLExecutionResult) = task {
-                    let gqlResponse =
-                        match result.Content with
-                        | Direct (data, errs) -> GQLResponse.Direct (result.DocumentId, data, errs)
-                        | RequestError (problemDetailsList) -> GQLResponse.RequestError (result.DocumentId, problemDetailsList)
-                        | _ ->
-                            GQLResponse.RequestError (
-                                result.DocumentId,
-                                [
-                                    GQLProblemDetails.Create ("subscriptions are not supported here (use the websocket endpoint instead).")
-                                ]
+    let rec private moduleType = getModuleType <@ moduleType @>
+
+    let ofTaskIResult ctx (taskRes: Task<IResult>) : HttpFuncResult = task {
+        let! res = taskRes
+        do! res.ExecuteAsync(ctx)
+        return Some ctx
+    }
+
+    let ofTaskIResult2 ctx (taskRes: Task<Result<IResult, IResult>>) : HttpFuncResult =
+        taskRes
+        |> TaskResult.defaultWith id
+        |> ofTaskIResult ctx
+
+    /// Set CORS to allow external servers (React samples) to call this API
+    let setCorsHeaders : HttpHandler =
+        setHttpHeader "Access-Control-Allow-Origin" "*"
+        >=> setHttpHeader "Access-Control-Allow-Headers" "content-type"
+
+    let private handleGraphQL<'Root> (next : HttpFunc) (ctx : HttpContext) =
```

Comment:
shouldn't `next : HttpFunc` be used somewhere?

Reply:
No it is implemented the same as Giraffe native extension method `WriteBytesAsync`